Clustering update

trevorcampbell opened the pull request:
Major overhaul of the clustering section.

ttimbers commented:
Some comments:
- Could we be a bit more explicit that clustering is a tool to answer exploratory statistical questions. And as such, the patterns we see are not extended to future cases (as in prediction) or generalized to the larger population (as in inference), but instead restricted to conclusions about the data in the dataset the clustering was performed on? But, as you already mention, useful for generating hypotheses/idea/new questions that could inform asking and answering additional predictive or inferential questions. Currently the term "exploratory" is only used in the overview section for the chapter.

ttimbers commented:
Also, we might want to add a section discussing the `nstart` argument of `kmeans` as we use it in the tutorial (and maybe the worksheet)?

ttimbers commented:
Last year, because R's outputs gives you tot sum of squares and between sum of squares, the students asked what these things were... However, another reason that might have come up was that a MDS students (and a TA for this course) gave this as a guest lecture. And he talked about these in his lecture slides... So my advice is either don't talk about these things anywhere, or explain them all...

trevorcampbell commented on 09-clustering.Rmd:
````diff
@@ -76,41 +122,323 @@ ggplot(data, aes(y = loyalty, x = csat, color = cluster)) +
+These two steps are repeated until the cluster assignments no longer change.
+For example, in the customer data example from earlier, our initialization might look like this:
+<center>
+```{r 10-toy-kmeans-init, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 4.35, fig.cap = "Random initialization of labels."}
````
I don't see the package loading message you're referring to. Screenshot??

trevorcampbell commented:
> Also, we might want to add a section discussing the `nstart` argument of `kmeans` as we use it in the tutorial (and maybe the worksheet)?

Done!

trevorcampbell commented on 09-clustering.Rmd:
````diff
@@ -76,41 +122,323 @@ ggplot(data, aes(y = loyalty, x = csat, color = cluster)) +
+clusters, we see that the decrease in total WSSD levels off (or forms an "elbow shape") when we reach roughly 
+the right number of clusters.
+<center>
+```{r 10-toy-kmeans-elbow, echo = FALSE, warning = FALSE, fig.height = 6, fig.width = 10, fig.cap = "Total WSSD for # clusters ranging from 1 to 9."}
````
I'll leave this as a TODO in the issues.

trevorcampbell commented on 09-clustering.Rmd:
````diff
@@ -76,41 +122,323 @@ ggplot(data, aes(y = loyalty, x = csat, color = cluster)) +
+clustering of our customer satisfaction and loyalty data for a number of clusters
+ranging from 1 to 9.
+
+```{r 10-toy-kmeans-vary-k, echo = FALSE, warning = FALSE, fig.height = 10, fig.width = 12, fig.cap = "Clustering of the customer data for # clusters ranging from 1 to 9."}
````
Done for both. NB Lecture slides will still have Xs for this round; TODO for next round.

trevorcampbell commented on 09-clustering.Rmd:
````diff
@@ -76,41 +122,323 @@ ggplot(data, aes(y = loyalty, x = csat, color = cluster)) +
+where the left column depicts the center update, 
+and the right column depicts the reassignment of data to clusters): 
+<center>
+```{r 10-toy-kmeans-iter, echo = FALSE, warning = FALSE, fig.height = 4, fig.width = 8}
````
Done

trevorcampbell commented on 09-clustering.Rmd:
```diff
@@ -76,41 +122,323 @@ ggplot(data, aes(y = loyalty, x = csat, color = cluster)) +
 
-## K-means clustering in R
+The second step is to add up the squared distance between each point in the cluster and the cluster center.
+We use the usual distance formula that we learned about in the classification chapter.
```
done

trevorcampbell commented on 09-clustering.Rmd:
```diff
@@ -76,41 +122,323 @@ ggplot(data, aes(y = loyalty, x = csat, color = cluster)) +
+the "quality" of a clustering, and what does it mean to improve it? 
+In K-means clustering, we measure the quality of a cluster by its
+*within-cluster sum-of-squared-distances* (WSSD). Computing this involves two steps:
+We first compute the mean of each variable we are using to cluster the data
```
done!

trevorcampbell commented:
> Could we be a bit more explicit that clustering is a tool to answer exploratory statistical questions. And as such, the patterns we see are not extended to future cases (as in prediction) or generalized to the larger population (as in inference), but instead restricted to conclusions about the data in the dataset the clustering was performed on? But, as you already mention, useful for generating hypotheses/idea/new questions that could inform asking and answering additional predictive or inferential questions. Currently the term "exploratory" is only used in the overview section for the chapter.

Clustering can actually help with prediction and other downstream tasks. Imagine a dataset that looks like a "cross" -- if we split the data using clustering and then do regression on each cluster separately, we might be a lot better at predicting things than if we tried to do a single regression on the whole dataset.

But since this course only uses clustering for exploration, I added a remark about that at the beginning.
